### DISCUSSION TITLE
Migration to apollo-compiler new beta version

### DISCUSSION THREAD

**Discussion by yanns** [apollographql/apollo-rs#736] [Show and tell]
We've migrated from apollo-compiler `0.11.3` to `1.0.0-beta.4`, and we are observing much better performances:

<img width="621" alt="Screenshot 2023-11-10 at 14 17 58" src="https://github.com/apollographql/apollo-rs/assets/51669/ada950b3-5f94-4c54-8898-5083cecaf3e0">

Congrats to all involved!

**Comment by yanns** [comment]
Updating to `1.0.0-beta.6` gives even better results:

<img width="488" alt="Screenshot 2023-11-10 at 15 44 10" src="https://github.com/apollographql/apollo-rs/assets/51669/787dd078-f618-441d-ad31-b3df36961314">

**Comment by SimonSapin** [comment]
Awesome! Can you say more about what this is measuring?

**Reply by yanns** [reply]
Yes this is the metric `apollo_router_processing_time` from the router.

I also measure the performance from the outside, and there is the same trend:

<img width="551" alt="Screenshot 2023-11-11 at 11 36 37" src="https://github.com/apollographql/apollo-rs/assets/51669/d2978722-e7f5-4309-be18-f91914613ad3">

**Reply by abernix** [reply]
@yanns Do you have particularly large operations or a particularly large schema?

**Reply by yanns** [reply]
Yes a [very large schema](https://github.com/commercetools/commercetools-api-reference/blob/main/api-specs/graphql/schema.sdl).
